src/openms/source/METADATA/ID/IdentificationData.cpp

```diff
@@ -251,154 +290,190 @@ namespace OpenMS
-    }
-    if ((parent.coverage < 0.0) || (parent.coverage > 1.0))
+    if (!no_checks_)
     {
-      String msg = "parent molecule coverage must be between 0 and 1";
-      throw Exception::IllegalArgument(__FILE__, __LINE__,
-                                       OPENMS_PRETTY_FUNCTION, msg);
+      if (parent.accession.empty())
+      {
+        String msg = "missing accession for parent sequence";
+        throw Exception::IllegalArgument(__FILE__, __LINE__,
+                                         OPENMS_PRETTY_FUNCTION, msg);
+      }
+      if ((parent.coverage < 0.0) || (parent.coverage > 1.0))
+      {
+        String msg = "parent sequence coverage must be between 0 and 1";
+        throw Exception::IllegalArgument(__FILE__, __LINE__,
+                                         OPENMS_PRETTY_FUNCTION, msg);
+      }
     }
 
-    return insertIntoMultiIndex_(parent_molecules_, parent,
-                                 parent_molecule_lookup_);
+    return insertIntoMultiIndex_(parents_, parent,
+                                 parent_lookup_);
   }
 
 
-  void IdentificationData::registerParentMoleculeGrouping(
-    const ParentMoleculeGrouping& grouping)
+  void IdentificationData::registerParentGroupSet(const ParentGroupSet& groups)
   {
-    checkAppliedProcessingSteps_(grouping.steps_and_scores);
-
-    for (const auto& group : grouping.groups)
+    if (!no_checks_)
     {
-      checkScoreTypes_(group.scores);
+      checkAppliedProcessingSteps_(groups.steps_and_scores);
 
-      for (const auto& ref : group.parent_molecule_refs)
+      for (const auto& group : groups.groups)
       {
-        if (!isValidHashedReference_(ref, parent_molecule_lookup_))
+        checkScoreTypes_(group.scores);
+
+        for (const auto& ref : group.parent_refs)
         {
-          String msg = "invalid reference to a parent molecule - register that first";
-          throw Exception::IllegalArgument(__FILE__, __LINE__,
-                                           OPENMS_PRETTY_FUNCTION, msg);
+          if (!isValidHashedReference_(ref, parent_lookup_))
+          {
+            String msg = "invalid reference to a parent sequence - register that first";
+            throw Exception::IllegalArgument(__FILE__, __LINE__,
+                                             OPENMS_PRETTY_FUNCTION, msg);
+          }
         }
       }
     }
 
-    parent_molecule_groupings_.push_back(grouping);
+    parent_groups_.push_back(groups);
 
     // add the current processing step?
     if ((current_step_ref_ != processing_steps_.end()) &&
-        (grouping.steps_and_scores.get<1>().find(current_step_ref_) ==
-         grouping.steps_and_scores.get<1>().end()))
+        (groups.steps_and_scores.get<1>().find(current_step_ref_) ==
+         groups.steps_and_scores.get<1>().end()))
     {
-      parent_molecule_groupings_.back().steps_and_scores.push_back(
+      parent_groups_.back().steps_and_scores.push_back(
```

Comment:
this is... like many other functions - not thread safe.
Is this documented? e.g., that users need to ensure non-concurrent modification of the datastructure?

Reply:
I've added a warning to the top-level class documentation.